### PR TITLE
[FRE-1211] add ledger warning for ethermint chains

### PR DESCRIPTION
**Pull request by codingki**
image

### CODE REVIEW

File: packages/widget/src/hooks/useShowCosmosLedgerWarning.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  })
+
+  return useMemo(() => {
+    if (!sourceAsset?.chainID) return false;
+    if (chainType !== "cosmos") return false;
+    if (!knownEthermintLikeChains.includes(sourceAsset?.chainID)) return false;
+    return !!cosmosAccount?.isNanoLedger
+  }, [
+    sourceAsset?.chainID,
+    chainType,
```
Comment (ericHgorski) [inline packages/widget/src/hooks/useShowCosmosLedgerWarning.ts:23]:
```suggestion
    chainType,
    cosmosAccount?.isNanoLedger,
```

File: packages/widget/src/pages/ErrorPage/ErrorPageCosmosLedgerWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          icon: ICONS.thinArrow,
+          onClick: onClickBack,
+        }}
+      />
+      <ErrorPageContent
+        title={`Warning: Ledger not supported`}
+        description={
+          <>
+            <SmallText color={theme.error.text} textAlign="center" textWrap="balance">
+              We currently do not support Ledger on Ethermint-based chains (such as Injective, Dymension, EVMOS, and similar networks).
```
Comment (ericHgorski) [inline packages/widget/src/pages/ErrorPage/ErrorPageCosmosLedgerWarning.tsx:33]:
```suggestion
              Ledger isn't currently supported on Ethermint chains (such as Injective, Dymension, EVMOS, and similar networks).
```

File: packages/widget/src/state/types.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export enum RoutePreference {
+  FASTEST = "Fastest",
+  CHEAPEST = "Cheapest",
+}
```
Comment (ericHgorski) [inline packages/widget/src/state/types.ts:4]: 👍🏼

Comment (codingki) [reply packages/widget/src/state/types.ts:4]: this is preventing circular dependancy

**Review by ericHgorski — APPROVED** [review]
lgtm

**Pull request merged**